pkg/kubectl/kubectl.go

```diff
@@ -44,6 +46,32 @@ func makeImageList(spec *api.PodSpec) string {
 	return strings.Join(listOfImages(spec), ",")
 }
 
+func NewThirdPartyResourceMapper(gvs []unversioned.GroupVersion, gvks []unversioned.GroupVersionKind) (meta.RESTMapper, error) {
+	mapper := meta.NewDefaultRESTMapper(gvs, func(gv unversioned.GroupVersion) (*meta.VersionInterfaces, error) {
+		for ix := range gvs {
+			if gvs[ix].Group == gv.Group && gvs[ix].Version == gv.Version {
+				return &meta.VersionInterfaces{
+					Codec:            api.Codec,
+					ObjectConvertor:  api.Scheme,
+					MetadataAccessor: meta.NewAccessor(),
+				}, nil
+			}
+		}
+		g, _ := latest.Group("metrics")
+		groupVersions := []string{}
+		if g.GroupVersions != nil {
```

Comment:
can't `g` be nil here if you skip the error?  Also, ranging over a nil slice or map does nothing, right?